django_mongodb_backend/forms/fields/embedded_model_array.py

```diff
@@ -0,0 +1,76 @@
+from django import forms
+from django.core.exceptions import ValidationError
+from django.forms import formset_factory, model_to_dict
+from django.forms.models import modelform_factory
+from django.utils.html import format_html, format_html_join
+
+
+def models_to_dicts(models):
+    """
+    Convert initial data (which is a list of model instances or None) to a
+    list of dictionary data suitable for a formset.
+    """
+    return [model_to_dict(model) for model in models or []]
+
+
+class EmbeddedModelArrayField(forms.Field):
+    def __init__(self, model, prefix, max_num=None, extra_forms=3, **kwargs):
+        self.model = model
+        self.prefix = prefix
+        self.formset = formset_factory(
+            form=modelform_factory(model, fields="__all__"),
+            can_delete=True,
+            max_num=max_num,
+            extra=extra_forms,
+            validate_max=True,
+        )
+        kwargs["widget"] = EmbeddedModelArrayWidget()
+        super().__init__(**kwargs)
+
+    def clean(self, value):
+        if not value:
+            return []
+        formset = self.formset(value, prefix=self.prefix)
+        if not formset.is_valid():
+            raise ValidationError(formset.errors + formset.non_form_errors())
+        cleaned_data = []
+        for data in formset.cleaned_data:
+            # The fallback to True skips empty forms.
+            if data.get("DELETE", True):
+                continue
+            data.pop("DELETE")  # The "delete" checkbox isn't part of model data.
```

Comment:
Don't know if it is more clear:
```suggestion
            if data.pop("DELETE", True):
                continue
```